taxonomy-manager-rest-server/src/main/java/com/digirati/taxman/rest/server/analysis/TextAnalyzer.java

```diff
@@ -0,0 +1,58 @@
+package com.digirati.taxman.rest.server.analysis;
+
+import com.digirati.taxman.common.rdf.RdfModelException;
+import com.digirati.taxman.common.rdf.RdfModelFactory;
+import com.digirati.taxman.common.taxonomy.CollectionModel;
+import com.digirati.taxman.common.taxonomy.ConceptModel;
+import com.digirati.taxman.rest.analysis.TextAnalysisInput;
+import com.digirati.taxman.rest.server.taxonomy.ConceptModelRepository;
+import com.digirati.taxonomy.manager.lookup.TextLookupService;
+import org.apache.jena.vocabulary.SKOS;
+
+import javax.ws.rs.WebApplicationException;
+import java.net.URI;
+
+public class TextAnalyzer {
+    private final TextLookupService lookupService;
+    private final RdfModelFactory modelFactory;
+    private final ConceptModelRepository concepts;
+
+    public TextAnalyzer(TextLookupService lookupService, RdfModelFactory modelFactory, ConceptModelRepository concepts) {
+        this.lookupService = lookupService;
+        this.modelFactory = modelFactory;
+        this.concepts = concepts;
+    }
+
+    /**
+     * Run the auto-tagger implementation on the given {@code input} and return a collection of {@link ConceptModel}s that
+     * are found in the input.
+     *
+     * @param input The document to tag.
+     * @return A list of {@link ConceptModel}s appearing as tags.
+     */
+    public CollectionModel tagDocument(TextAnalysisInput input) {
+        var ctx = lookupService.search(input.getText());
+        var matches = ctx.getMatchedConcepts();
+
+        try {
+            var builder = modelFactory.createBuilder(CollectionModel.class);
+            builder.setUri(URI.create("urn:collection"));
+
+            var matchedConcepts = matches.parallelStream()
+                    .flatMap(match -> match.getConceptIds().stream())
+                    .map(concepts::find);
```

Comment:
I don't think we should make a separate call to get each concept - probably worth adding a db function that takes a set of UUIDs and gives back the relevant concepts.

Reply:
This one is a little odd. It doesn't seem to be possible to get PostgreSQL to use the unique UUID index when we're querying for a collection of concepts (here's the plan: http://tatiyants.com/pev/#/plans/plan_1562595043043).

Instead, it does a sequential scan. I suppose this is down to 1 of 2 reasons: 

1) the query planner is unable to optimize the `ANY(_uuids)` condition. 
2) the query planner can optimize it and decides a sequential scan is faster.

At a large number of concepts that sequential scan starts to become a problem (especially with wide tables containing JSON), but it may turn out that index scans start kicking in at a high enough volume. I'll push an implementation that uses the array approach and we can go with that until it becomes an issue.